lib/rust/parser/src/syntax/statement/function_def.rs

```diff
@@ -87,7 +88,15 @@ pub fn parse_constructor_definition<'s>(
         }))
     }
     let (name, inline_args) = parse_constructor_decl(items, start, precedence, args_buffer);
-    Tree::constructor_definition(name, inline_args, block_args)
+    let private =
+        (private_keywords_start < start).then(|| into_private_keyword(items.pop().unwrap()));
+    Tree::constructor_definition(private, name, inline_args, block_args)
+}
+
+fn into_private_keyword(item: Item) -> token::PrivateKeyword {
+    let Item::Token(keyword) = item else { unreachable!() };
+    let token::Variant::PrivateKeyword(variant) = keyword.variant else { unreachable!() };
+    keyword.with_variant(variant)
```

Comment:
The same function is in `statement.rs`

Reply:
Implemented some downcasting helpers so we don't need a special function for this case.